tracer/src/Datadog.Trace/RemoteConfigurationManagement/Product.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="Product.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.RemoteConfigurationManagement
+{
+    internal abstract class Product
+    {
+        protected Product()
+        {
+            AppliedConfigurations = new Dictionary<string, RemoteConfigurationCache>();
+        }
+
+        public event EventHandler<ProductConfigChangedEventArgs> ConfigChanged;
+
+        public abstract string Name { get; }
+
+        public Dictionary<string, RemoteConfigurationCache> AppliedConfigurations { get; }
+
+        public void AssignConfigs(List<RemoteConfiguration> changedConfigs)
+        {
+            var configurations =
+                changedConfigs
+                   .Where(RemoteConfigurationPredicate)
+                   .Select(configuration => configuration.Contents)
+                   .ToList()
```

Comment:
I agree with you, there are a too much `ToList()` calls.
However, at this specific example, I prefer to avoid invoking event, if no configurations are exists, and I don't want multiple enumerations, so I don't know exactly, how to resolve this.